client/my-sites/plans/current-plan/trials/use-go-to-checkout-with-plan.tsx

```diff
@@ -0,0 +1,59 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import {
+	PLAN_BUSINESS,
+	PLAN_ECOMMERCE_MONTHLY,
+	PLAN_ECOMMERCE_TRIAL_MONTHLY,
+	PLAN_WOOEXPRESS_MEDIUM_MONTHLY,
+} from '@automattic/calypso-products';
+import page from '@automattic/calypso-router';
+import { getTrialCheckoutUrl } from 'calypso/lib/trials/get-trial-checkout-url';
+import { useSelector } from 'calypso/state';
+import { getSelectedPurchase, getSelectedSite } from 'calypso/state/ui/selectors';
+import useOneDollarOfferTrack from '../../hooks/use-onedollar-offer-track';
+
+const getTargetPlanAndTrackEvent = ( isEcommerceTrial: boolean, isWooExpressTrial: boolean ) => {
+	if ( isEcommerceTrial ) {
+		return isWooExpressTrial
+			? [ PLAN_WOOEXPRESS_MEDIUM_MONTHLY, 'calypso_wooexpress_my_plan_cta' ]
+			: [ PLAN_ECOMMERCE_MONTHLY, 'calypso_entrepreneur_my_plan_cta' ];
+	}
+
+	return [ PLAN_BUSINESS, 'calypso_migration_my_plan_cta' ];
+};
+
+const useGoToCheckoutWithPlan = () => {
+	const selectedSite = useSelector( getSelectedSite );
+	const purchase = useSelector( getSelectedPurchase );
+
+	const currentPlanSlug = selectedSite?.plan?.product_slug ?? '';
+	const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
+	const isWooExpressTrial = purchase?.isWooExpressTrial;
```

Comment:
This question comes from the background I am missing and curiosity: Why are we getting these two values differently, i.e. why we cannot get them the same way? 🤔 

I understand that there's a chance the answer might not be straightforward so we can discuss it in chat or a call.

Reply:
Both Entrepreneur and Woo Express trial share the same product, `PLAN_ECOMMERCE_TRIAL_MONTHLY`, and on differ on billing plans.